Tests/ClientTests/OnboardingTests/OnboardingButtonActionTests.swift

```diff
@@ -0,0 +1,150 @@
+
+        XCTAssertNil(mockDelegate.action)
+    }
+
+    func testsubject_whenOnlyOneButtonExists_secondaryButtonIsHidden() {
+        setSubjecUpWith(firstAction: .nextCard, twoButtons: false)
+
+        subject.secondaryAction()
+
+        XCTAssertNil(mockDelegate.action)
+    }
+
+    func testsubject_primaryAction_returnsNextCardAction() {
+        setSubjecUpWith(firstAction: .nextCard)
+
+        subject.primaryAction()
+
+        XCTAssertEqual(mockDelegate.action, OnboardingActions.nextCard)
+    }
+
+    func testsubject_buttonAction_returnsPrivacyPolicyAction() {
+        setSubjecUpWith(firstAction: .readPrivacyPolicy)
+
+        subject.linkButtonAction()
+
+        XCTAssertEqual(mockDelegate.action, OnboardingActions.readPrivacyPolicy)
+    }
+
+    func testsubject_buttonAction_returnsNotifiactionsAction() {
+        setSubjecUpWith(firstAction: .requestNotifications)
+
+        subject.primaryAction()
+
+        XCTAssertEqual(mockDelegate.action, OnboardingActions.requestNotifications)
+    }
+    func testsubject_buttonAction_returnsSyncAction() {
+        setSubjecUpWith(firstAction: .syncSignIn)
+
+        subject.primaryAction()
+
+        XCTAssertEqual(mockDelegate.action, OnboardingActions.syncSignIn)
+    }
+
+    func testsubject_buttonAction_returnsSetAsDefaultAction() {
+        setSubjecUpWith(firstAction: .setDefaultBrowser)
+
+        subject.primaryAction()
+
+        XCTAssertEqual(mockDelegate.action, OnboardingActions.setDefaultBrowser)
+    }
+
+    // MARK: - Helpers
+    func setSubjecUpWith(
+        firstAction: OnboardingActions,
+        twoButtons: Bool = true
+    ) {
+        var buttons: OnboardingButtons
+        if twoButtons {
+            buttons = OnboardingButtons(
+                primary: OnboardingButtonInfoModel(
+                    title: .Onboarding.Sync.SignInAction,
+                    action: firstAction),
+                secondary: OnboardingButtonInfoModel(
+                    title: .Onboarding.Sync.SkipAction,
+                    action: .nextCard))
+        } else {
+            buttons = OnboardingButtons(
+                primary: OnboardingButtonInfoModel(
+                    title: .Onboarding.Sync.SignInAction,
+                    action: firstAction))
+        }
+
+        let mockInfoModel = OnboardingCardInfoModel(
+            name: "signSync",
+            title: String(format: .Onboarding.Sync.Title),
+            body: String(format: .Onboarding.Sync.Description),
+            link: nil,
+            buttons: buttons,
+            type: .freshInstall,
+            a11yIdRoot: AccessibilityIdentifiers.Onboarding.signSyncCard,
+            imageID: ImageIdentifiers.onboardingSyncv106)
+        let mockCardViewModel = LegacyOnboardingCardViewModel(
+            cardType: .welcome,
+            infoModel: mockInfoModel,
+            isFeatureEnabled: true)
+
+        mockDelegate = MockIntroViewController()
+        subject = OnboardingCardViewController(
```

Comment:
Bonus, we can track leaks with `trackForMemoryLeaks` easily since we have the subject easily available in the setup function